File: dfs_bfs_backtracking/python/b2606.py
N = int(input())
connects = int(input())
connections = [[] for _ in range(N + 1)]


def DFS(connections, N):
    cnt = 0
    visited = [0] * (N + 1)
    check_stack = [1]
    visited[1] = 1
    while check_stack:
        cnt += 1
        temp_node = check_stack.pop()
        for other in connections[temp_node]:
            if not visited[other]:
                check_stack.append(other)
                visited[other] = 1
    return cnt - 1


for i in range(connects) :
    start, end = map(int, input().split())
    connections[start].append(end)
    connections[end].append(start)
print(DFS(connections, N))

File: dfs_bfs_backtracking/cpp/b2606.cpp
#include<iostream>
#include<vector>
#include<stack>

std::vector<std::vector<int>> connections;

int count_infection(std::vector<std::vector<int>>& connections) {
	int cnt = 0;
	int infected[101]{ 0 };

	std::stack<int> check_stack;
	check_stack.push(1);
	infected[1] = 1;

	while (check_stack.size() > 0) {
		cnt += 1;
		int check_node = check_stack.top();
		check_stack.pop();
		for (int& connect : connections[check_node]) {
			if (!infected[connect]) {
				check_stack.push(connect);
				infected[connect] = 1;
			}
		}
	}
	return cnt - 1;
}

int main() {
	int N;
	std::cin >> N;
	connections.resize(N + 1);

	int connect;
	std::cin >> connect;
	for (int i = 0; i < connect; ++i) {
		int temp_a, temp_b;
		std::cin >> temp_a >> temp_b;
		connections[temp_a].push_back(temp_b);
		connections[temp_b].push_back(temp_a);
	}
	
	std::cout << count_infection(connections) << std::endl;
}